framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -139,6 +139,28 @@ public Constraint pop() {
     return list.remove(0);
   }
 
+  /**
+   * Adds the constraint to the beginning of this set.
+   *
+   * @param constraint a constraint
+   */
+  public void push(Constraint constraint) {
+    if (constraint != null && !list.contains(constraint)) {
+      list.add(0, constraint);
+    }
+  }
+
+  /**
+   * Adds the constraints to the beginning of this set and matatines the order of the constraints.
+   *
+   * @param constraints constraints
+   */
+  public void push(ConstraintSet constraints) {
+    for (int i = constraints.list.size() - 1; i > -1; i--) {
```

Comment:
Minor: unless the list is a linked list or very few items are being pushed, this is probably an inefficient way to add multiple items to the front.